app/src/main/java/com/duckduckgo/app/fire/FireAnimationLoader.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2020 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.fire
+
+import android.content.Context
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+import com.airbnb.lottie.LottieCompositionFactory
+import com.duckduckgo.app.browser.R
+import com.duckduckgo.app.statistics.VariantManager
+import javax.inject.Inject
+
+class FireAnimationLoader @Inject constructor(
+    private val variantManager: VariantManager,
+    private val context: Context
+) : LifecycleObserver {
+
+    @OnLifecycleEvent(Lifecycle.Event.ON_START)
+    fun onAppStart() {
+        if (variantManager.getVariant().hasFeature(VariantManager.VariantFeature.FireButtonEducation)) {
```

Comment:
This is going to cause problems for detecting installation referrer, as it's eagerly allocating a variant before we can detect and assign a variant based on referrer.

Could we defer initializing the animation until `BrowserActivity`, or at least until after `LaunchViewModel.waitForReferrerData()` has finished?

Reply:
Good catch, I didn't think about the referrer. For simplicity, I'm going to move it to `BrowserActivity`, and leave a comment saying that it can be moved back to the Application lifecycle once we have experiment results, as at that point we will remove the variant manager dependency.